components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/test/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOTest.java

```diff
@@ -304,15 +312,15 @@ public void testGetRolesWithFilter() throws Exception {
         identityUtil.when(IdentityUtil::getMaximumItemPerPage)
                 .thenReturn(IdentityCoreConstants.DEFAULT_MAXIMUM_ITEMS_PRE_PAGE);
 
-        addRole("role1", APPLICATION_AUD, "test-app-id", roleDAO);
-        addRole("roleA2", APPLICATION_AUD, "test-app-id", roleDAO);
-        addRole("role3", ORGANIZATION_AUD, "test-org-id", roleDAO);
+        addRole(roleNamesList.get(0), APPLICATION_AUD, SAMPLE_APP_ID, roleDAO);
+        addRole(roleNamesList.get(1), APPLICATION_AUD, SAMPLE_APP_ID, roleDAO);
+        addRole(roleNamesList.get(2), ORGANIZATION_AUD, SAMPLE_ORG_ID, roleDAO);
 
         List<String> expectedRoles = new ArrayList<>();
-        expectedRoles.add("roleA2");
+        expectedRoles.add(roleNamesList.get(1));
 
         List<ExpressionNode> expressionNodes =
-                getExpressionNodes("name co roleA and audience co application and audienceId co test-app-id");
+                getExpressionNodes("name co 2 and audience co application and audienceId co test-app-id");
```

Comment:
Looks like mistake while refactoring.

Reply:
This was changed to "2" as the role name was changed from "roleA2" to "role2" for consistency. Also, this will not affect other test cases as they create/update/assign/delete roles independently.